CMakeLists.txt

```diff
@@ -0,0 +1,169 @@
+cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
+
+project(KokkosTools CXX)
+
+# Set policies
+cmake_policy(SET CMP0054 NEW) # simplify if()
+cmake_policy(SET CMP0057 NEW) # support IN_LIST in if()
+cmake_policy(SET CMP0111 NEW) # error if library not found
+
+# Disable in-source builds to prevent source tree corruption.
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_BINARY_DIR)
+  message(FATAL_ERROR "FATAL: In-source builds are not allowed. You should create a separate directory for build files.")
+endif()
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+message(STATUS)
+message(STATUS Configuring Kokkos-Tools)
+message(STATUS)
+
+# Tools settings
+set(KOKKOSTOOLS_ENABLE_CALIPER ON CACHE BOOL "Enable building Caliper library")
+set(KOKKOSTOOLS_ENABLE_APEX    ON CACHE BOOL "Enable building Apex library")
+
+# Check Kokkos settings
+if(TARGET Kokkos::kokkos)
+  message(STATUS "Using Kokkos settings for KokkosTools (source: ${Kokkos_SOURCE_DIR})")
+else()
+  message(WARNING "Not using Kokkos for tools configuration\n\tNote: using settings from installed is Kokkos not yet supported")
```

Comment:
Oh, this is a really slick idea

Reply:
Why would there be a `Kokkos::Kokkos` target w/o a `find_package(Kokkos)`?

Reply:
In my app's CMakeLists.txt

```
add_subdirectory(kokkos) # defines the target
add_subdirectory(kokkos-tools)
```

Reply:
That's not very clear, especially with the `set(CMAKE_CXX_COMPILER ${Kokkos_CXX_COMPILER})` down below.

Reply:
Yeah, could definitely use a comment. It's also more of a Lawrence Livermore-ism than something Sandia codes do today

Reply:
I cleaned this up a bit (excuse messy draft). The idea is that:
1. we check if Kokkos is built within same parent project - and before Tools, as in David's example above. Is there a better way than checking for `TARGET Kokkos::kokkos` ? In this case we're all set up with options, compiler and flags.
2. we check for installed Kokkos in `Kokkos_ROOT` and extract all relevant settings from there.
3. in case Kokkos is nowhere found, we might just built Tools alone (figuring reasonable option defaults) or error out.

Any suggestions appreciated (where's the slicky part ?)